exetera/core/session.py

```diff
@@ -652,42 +652,82 @@ def create_like(self, field, dest_group, dest_name, timestamp=None, chunksize=No
+            return group.create_fixed_string(name, length, timestamp, chunksize)
 
 
     def create_categorical(self, group, name, nformat, key,
                            timestamp=None, chunksize=None):
-        if isinstance(group, ds.Dataset):
-            pass
-        elif isinstance(group, df.DataFrame):
-            return group.create_categorical(self, name, nformat,key,timestamp,chunksize)
+        if not isinstance(group, (df.DataFrame, h5py.Group)):
+            if isinstance(group, ds.Dataset):
+                raise ValueError("'group' must be an ExeTera DataFrame rather than a"
+                                 " top-level Dataset")
+            else:
+                raise ValueError("'group' must be an Exetera DataFrame but a "
+                                 "{} was passed to it".format(type(group)))
+
+        if isinstance(group, h5py.Group):
+            return fld.categorical_field_constructor(self, group, name, timestamp, chunksize)
+        else:
+            return group.create_categorical(name, nformat, key, timestamp, chunksize)
 
 
     def create_numeric(self, group, name, nformat, timestamp=None, chunksize=None):
-        if isinstance(group,ds.Dataset):
-            pass
-        elif isinstance(group,df.DataFrame):
-            return group.create_numeric(self,name, nformat, timestamp, chunksize)
+        if not isinstance(group, (df.DataFrame, h5py.Group)):
+            if isinstance(group, ds.Dataset):
+                raise ValueError("'group' must be an ExeTera DataFrame rather than a"
+                                 " top-level Dataset")
+            else:
+                raise ValueError("'group' must be an Exetera DataFrame but a "
+                                 "{} was passed to it".format(type(group)))
 
+        if isinstance(group, h5py.Group):
+            return fld.numeric_field_constructor(self. group, name, timestamp, chunksize)
+        else:
+            return group.create_numeric(name, nformat, timestamp, chunksize)
 
 
     def create_timestamp(self, group, name, timestamp=None, chunksize=None):
-        if isinstance(group,ds.Dataset):
-            pass
-        elif isinstance(group,df.DataFrame):
-            return group.create_timestamp(self,name, timestamp, chunksize)
+        if not isinstance(group, (df.DataFrame, h5py.Group)):
+            if isinstance(group, ds.Dataset):
+                raise ValueError("'group' must be an ExeTera DataFrame rather than a"
+                                 " top-level Dataset")
+            else:
+                raise ValueError("'group' must be an Exetera DataFrame but a "
+                                 "{} was passed to it".format(type(group)))
+
+        if isinstance(group, h5py.Group):
+            return fld.categorical_field_constructor(self, group, name, timestamp, chunksize)
+        else:
+            return group.create_timestamp(name, timestamp, chunksize)
 
 
+    #TODO: update this method for the existence of dataframes
```

Comment:
TODO is a no-op as this code works for both groups and dataframes; remove the comment